src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts

```diff
@@ -26,6 +30,12 @@ export class RankingListComponent implements OnInit {
   ngDoCheck() {
     if (this.hiddenGems !== this.appComponent.hiddenGems) {
       this.hiddenGems = this.appComponent.hiddenGems;
```

Comment:
There's a lot more arrays now than there were before, it's obvious what's happening when you look at the html file but could you add a comment here or at the top of the class explaining briefly what the arrays are used for so flipping over the html file is not necessary